### PR TITLE
Change stable ci nuget build command

### PR DESCRIPTION
**Pull request by Ullmie02**
**Changes**
Another try at fixing nuget ci. This time I actually tested it.

- Don't specify `.snupkg` files in push because they get pushed automatically (https://github.com/Microsoft/azure-pipelines-tasks/issues/10140)
- Build Nuget packages with version specified in the git tag name (see Docker images) https://github.com/jellyfin/jellyfin/blob/406ae3e43a20216292c554151fa2d0e2a09edfa3/.ci/azure-pipelines-package.yml#L126-L128

Note: I'm tagging #4859 for backport as well since I forgot that at the time of making the PR

This is the sample output from a build for the tag `v10.7.1-rc4` with package names changed to `Example.*`

**Build task:**
```
...
Successfully created package '/home/vsts/work/1/a/Example.Naming.10.7.1-rc4.nupkg'.
Successfully created package '/home/vsts/work/1/a/Example.Naming.10.7.1-rc4.snupkg'.
...
Successfully created package '/home/vsts/work/1/a/Example.Data.10.7.1-rc4.nupkg'.
Successfully created package '/home/vsts/work/1/a/Example.Data.10.7.1-rc4.snupkg'.
...
Successfully created package '/home/vsts/work/1/a/Example.Common.10.7.1-rc4.nupkg'.
Successfully created package '/home/vsts/work/1/a/Example.Common.10.7.1-rc4.snupkg'.
...
Successfully created package '/home/vsts/work/1/a/Example.Controller.10.7.1-rc4.nupkg'.
Successfully created package '/home/vsts/work/1/a/Example.Controller.10.7.1-rc4.snupkg'.
...
Successfully created package '/home/vsts/work/1/a/Example.Model.10.7.1-rc4.nupkg'.
Successfully created package '/home/vsts/work/1/a/Example.Model.10.7.1-rc4.snupkg'.
```
**Push task:**
```
/usr/bin/mono /opt/hostedtoolcache/NuGet/5.4.0/x64/nuget.exe push /home/vsts/work/1/a/Example.Common.10.7.1-rc4.nupkg -NonInteractive -Source https://api.nuget.org/v3/index.json -ApiKey *** -ConfigFile /home/vsts/work/1/Nuget/tempNuGet_199.config -Verbosity Detailed
NuGet Version: 5.4.0.6315
Pushing Example.Common.10.7.1-rc4.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 794ms
Your package was pushed.
Pushing Example.Common.10.7.1-rc4.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 435ms
Your package was pushed.
/usr/bin/mono /opt/hostedtoolcache/NuGet/5.4.0/x64/nuget.exe push /home/vsts/work/1/a/Example.Controller.10.7.1-rc4.nupkg -NonInteractive -Source https://api.nuget.org/v3/index.json -ApiKey *** -ConfigFile /home/vsts/work/1/Nuget/tempNuGet_199.config -Verbosity Detailed
NuGet Version: 5.4.0.6315
Pushing Example.Controller.10.7.1-rc4.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 1085ms
Your package was pushed.
Pushing Example.Controller.10.7.1-rc4.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 475ms
Your package was pushed.
/usr/bin/mono /opt/hostedtoolcache/NuGet/5.4.0/x64/nuget.exe push /home/vsts/work/1/a/Example.Data.10.7.1-rc4.nupkg -NonInteractive -Source https://api.nuget.org/v3/index.json -ApiKey *** -ConfigFile /home/vsts/work/1/Nuget/tempNuGet_199.config -Verbosity Detailed
NuGet Version: 5.4.0.6315
Pushing Example.Data.10.7.1-rc4.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 794ms
Your package was pushed.
Pushing Example.Data.10.7.1-rc4.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 366ms
Your package was pushed.
/usr/bin/mono /opt/hostedtoolcache/NuGet/5.4.0/x64/nuget.exe push /home/vsts/work/1/a/Example.Model.10.7.1-rc4.nupkg -NonInteractive -Source https://api.nuget.org/v3/index.json -ApiKey *** -ConfigFile /home/vsts/work/1/Nuget/tempNuGet_199.config -Verbosity Detailed
NuGet Version: 5.4.0.6315
Pushing Example.Model.10.7.1-rc4.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 1021ms
Your package was pushed.
Pushing Example.Model.10.7.1-rc4.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 454ms
Your package was pushed.
/usr/bin/mono /opt/hostedtoolcache/NuGet/5.4.0/x64/nuget.exe push /home/vsts/work/1/a/Example.Naming.10.7.1-rc4.nupkg -NonInteractive -Source https://api.nuget.org/v3/index.json -ApiKey *** -ConfigFile /home/vsts/work/1/Nuget/tempNuGet_199.config -Verbosity Detailed
NuGet Version: 5.4.0.6315
Pushing Example.Naming.10.7.1-rc4.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Created https://www.nuget.org/api/v2/package/ 841ms
Your package was pushed.
Pushing Example.Naming.10.7.1-rc4.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...
  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 414ms
Your package was pushed.
```

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

**Comment by EraYaN** [comment]
Alright so this broke rc2 nuget publish, guess we'll publish those manually then.

**Review by EraYaN — APPROVED** [review]

File: .ci/azure-pipelines-package.yml [hunk 1]
```diff
@@ -193,20 +193,34 @@ jobs:
   pool:
     vmImage: 'ubuntu-latest'
 
+  variables:
+  - name: JellyfinVersion
+    value: 0.0.0
```
Comment (EraYaN) [inline .ci/azure-pipelines-package.yml:198]: You can set the value to a replace expression

```suggestion
    value: $[replace(variables['Build.SourceBranch'],'refs/tags/v','')]
```

just like what was done here: https://github.com/jellyfin/jellyfin-apiclient-java/blob/master/.ci/azure-pipelines.yml#L65

File: .ci/azure-pipelines-package.yml [hunk 2]
```diff
@@ -193,20 +193,34 @@ jobs:
...
   steps:
   - task: UseDotNet@2
     displayName: 'Use .NET 5.0 sdk'
     inputs:
       packageType: 'sdk'
       version: '5.0.x'
 
+  - script: echo "##vso[task.setvariable variable=JellyfinVersion]$( awk -F '/' '{ print $NF }' <<<'$(Build.SourceBranch)' | sed 's/^v//' )"
+    displayName: Set release version (stable)
+    condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/v')
```
Comment (EraYaN) [inline .ci/azure-pipelines-package.yml:209]:
```suggestion
```

**Review by EraYaN — APPROVED** [review]

**Review by joshuaboniface — APPROVED** [review]
# :rocket:

**Pull request merged**